src/main/java/crawlercommons/mimetypes/MimeTypeDetector.java

```diff
@@ -95,43 +104,65 @@ public MimeTypeDetector() {
         mimeTypes = new ArrayList<>();
 
         // Add all text patterns without and with a BOM.
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?xml"));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?xml", true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?XML"));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?XML", true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<!--"));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<!--", true));
-
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "http://"));
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "http://", true));
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "https://"));
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "https://", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?xml", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?XML", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<!--", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<urlset", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<sitemapindex", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rss", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<feed", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rdf", true, true));
+
+        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "http://", true, true));
+        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "https://", true, true));
 
         mimeTypes.add(new MimeTypeEntry(GZIP_MIMETYPES[0], "\037\213"));
         mimeTypes.add(new MimeTypeEntry(GZIP_MIMETYPES[0], 0x1F, 0x8B));
 
         maxPatternLength = 0;
         for (MimeTypeEntry entry : mimeTypes) {
-            maxPatternLength = Math.max(maxPatternLength, entry.getPattern().length);
+            int length = entry.getPattern().length;
```

Comment:
As per comment below, I think we can skip this by just treating BOM and leading whitespace checks as special, and adjust the offset for those, then fall into the regular pattern check.

Reply:
The method `detect(InputStream is)` requires the max. pattern length to create the byte[] array. The length must be longer for "text" patterns. But with a isTextPattern flag, this could be simplified by just adding LEADING_WHITESPACE_MAX_SKIP.